Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane is mapped to the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+def ℂ_ℤ := {z : ℂ // ¬ ∃ (n : ℤ), z = ↑n}
```

Comment:
Can we give this a name and make this a (scoped) notation? Then you could use it in the names below as well